docs/rules/isolated-functions.md

```diff
@@ -0,0 +1,303 @@
+
+### selectors
+
+Type: `string[]`\
+Default: `[]`
+
+Array of [ESLint selectors](https://eslint.org/docs/developer-guide/selectors) to identify isolated functions. Useful for custom naming conventions or framework-specific patterns.
+
+```js
+{
+	'unicorn/isolated-functions': [
+		'error',
+		{
+			selectors: [
+				'FunctionDeclaration[id.name=/lambdaHandler.*/]'
+			]
+		}
+	]
+}
+```
+
+### comments
+
+Type: `string[]`\
+Default: `['@isolated']`
+
+Array of comment strings that mark functions as isolated. Functions with inline, block, or JSDoc comments tagged with these strings will be considered isolated. (Definition of "tagged": either the comment consists solely of the tag, or starts with it, and has an explanation following a hyphen, like `// @isolated - this function will be stringified`).
+
+```js
+{
+	'unicorn/isolated-functions': [
+		'error',
+		{
+			comments: [
+				'@isolated',
+				'@remote'
+			]
+		}
+	]
+}
+```
+
+### globals
+
+Type: `object`\
+Default: `undefined` (uses ESLint's language options globals)
+
+Controls how global variables are handled. When not specified, uses ESLint's language options globals. When specified as an object, each key is a global variable name and the value controls its behavior:
+
+- `'readonly'`: Global variable is allowed but cannot be written to
+- `'writable'`: Global variable is allowed and can be read/written
+- `'off'`: Global variable is not allowed
+
+```js
+{
+	'unicorn/isolated-functions': [
+		'error',
+		{
+			globals: {
+				console: 'writable',     // Allowed and writable
+				fetch: 'readonly',       // Allowed but readonly
+				process: 'off'           // Not allowed
+			}
+		}
+	]
+}
+```
+
+## Examples
+
+### Custom function names
+
+```js
+{
+	'unicorn/isolated-functions': [
+		'error',
+		{
+			functions: [
+				'makeSynchronous',
+				'createWorker',
+				'serializeFunction'
+			]
+		}
+	]
+}
+```
+
+### Lambda function naming convention
+
+```js
+{
+	'unicorn/isolated-functions': [
+		'error',
+		{
+			selectors: [
+				'FunctionDeclaration[id.name=/lambdaHandler.*/]'
+			]
+		}
+	]
+}
+```
+
+```js
+const foo = 'hi';
+
+function lambdaHandlerFoo() { // ❌ Will be flagged as isolated
+	return foo.slice();
+}
+
+function someOtherFunction() { // ✅ Not flagged
+	return foo.slice();
+}
+
+createLambda({
+	name: 'fooLambda',
+	code: lambdaHandlerFoo.toString(), // Function will be serialized
+});
+```
+
+### Default behavior (using ESLint's language options)
+
+```js
+// Uses ESLint's language options globals by default
+makeSynchronous(async () => {
+	console.log('Starting...'); // ✅ Allowed if console is in language options
+	const response = await fetch('https://api.example.com'); // ✅ Allowed if fetch is in language options
+	return response.text();
+});
+```
+
+### Disallowing all globals
+
+```js
+{
+	'unicorn/isolated-functions': [
+		'error',
+		{
+			globals: {} // Empty object disallows all globals
```

Comment:
```suggestion
			globals: {} // Empty object disallows all globals except language globals
```

Reply:
Maybe we should mention that language globals are always enabled; they need to be disabled explicitly.

Reply:
Done. I think the behaviour is less confusing now - before `globals: {}` could turn off _everything_ in language options, and I don't think it was good for `globals: undefined` to be so dramatically different from `globals: {}`. 

Now, `overrideGlobals` does just that - it overrides globals that otherwise are defined, so if you want to turn off globals you turn them off explicitly.